### PR TITLE
Trapezoidal profile

### LINKED ISSUE #4: Simplify the SmoothControl command
There is a potential to simplify the SmoothControl command. I envision it being a single command:

moveTo(Position2D, Velocity)

How do we initialize the system? Should this happen through another ResetPosition(Position2D) command?

### LINKED ISSUE #5: Trapezoidal Velocity Control
We may need more velocity control in order to handle acceleration. Acceleration control may be needed in order to reduce wheel slip, but will be greatly used in order to handle turning and direction change

### PR DESCRIPTION
**Pull request by karthikappana35**
Purpose:
Add control over velocity during turning to allow the robot to speed up gradually instead of going at a constant velocity throughout turns
Changed Files:
SmoothControl.java - used additional formulas from the research paper that commands an altered velocity given a maximum velocity to add additional velocity control while making a turn. Reduces the velocity based on turn rate. 
DriveTo.java - modified the parameters to use a Position2d instead of a TargetPosition2d for simplicity purposes and added an additional parameter for the maximum velocity. Also added another parameter to change whether the robot is driving forwards or backward.
Constants.java - adjusted I gain to match tuned values and changed the closed-loop ramp constant to 2 to allow for smooth acceleration

### CODE REVIEW

**Comment by karthikappana35** [comment]
[10.1109/ICRA.2011.5980167](https://ieeexplore.ieee.org/document/5980167/) Research paper

**Review by mahisvivemulapalli — APPROVED** [review]
looks good

**Review by lillaantal1 — APPROVED** [review]

**Review by PranjalMhetre — APPROVED** [review]
Looks good!

**Review by NinaPlazonic — APPROVED** [review]

File: src/main/java/frc/robot/commands/DriveTo.java [hunk 1]
```diff
@@ -41,24 +43,32 @@ public void execute() {
...
+        }
 
-        try {
+        // Compute turn rate in radians and update range
+        double omegaDesired = m_smoothControl.computeTurnRate(m_kinematics.getPose(), m_targetPose, m_maxVelocity);
+        
+        if(true == m_inReverse){
             // Calculate vR in feet per second
-            vR = m_targetPose.getVelocity() + (TRACK_WIDTH_FEET / 2) * m_smoothControl.getTurnRateRadians();
+            vR = m_maxVelocity - (TRACK_WIDTH_FEET / 2) * omegaDesired;
```
Comment (bjmcternan) [inline src/main/java/frc/robot/commands/DriveTo.java]: vR and vL need to be negative if m_inReverse is true

**Review by bjmcternan — CHANGES_REQUESTED** [review]

**Review by bjmcternan — APPROVED** [review]

**Pull request merged**